src/options.rs

```diff
@@ -31,24 +32,30 @@ impl Options {
         }
     }
 
-    pub fn add_option_bool(&mut self, name: &str, value: bool) -> Result<(), ()> {
+    pub fn add_option_bool(&mut self, name: &str, value: bool) -> Result<()> {
         let name_c = CString::new(name).unwrap();
-        res_to_result(unsafe { extern_options::options_add_option_bool(self.ptr, name_c.as_ptr(), value) })
+        res_to_result(
+            unsafe { extern_options::options_add_option_bool(self.ptr, name_c.as_ptr(), value) }
+        ).or(Err(Error::APIError("Could not add a bool option")))
```

Comment:
Similarly here. The reason for this function to fail is that the options aren't locked. So the "Could not add a bool option" message doesn't really hint at why it failed, when I think that it would be better if it did.